src/envoy/utils/authn.cc

```diff
@@ -47,6 +48,11 @@ void Authentication::SaveAuthAttributesToStruct(
                 result.peer_user());
     setKeyValue(data, istio::utils::AttributeName::kSourcePrincipal,
                 result.peer_user());
+    std::string source_ns("");
+    if (GetSourceNamespace(result.peer_user(), &source_ns)) {
+      setKeyValue(data, istio::utils::AttributeName::kSourceNamespace,
```

Comment:
If namespace is extracted from certificate, should it be source.auth.namespace or smt like that to indicate it is authenticated?

Reply:
I tend to not to introduce a new attribute. User have to explicitly use `source.auth.namespace` which seems unnecessary considering we already have the similar `source.principal`. And if we have both `source.auth.namespace` and `source.namespace`, it makes it harder to use them at the same time.
@wattli @diemtvu @wenchenglu WDYT?